megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java

```diff
@@ -2513,10 +2513,11 @@ protected void okAction() {
             }
             CLIENT_PREFERENCES.setMapTileset(tileSetFileName);
         }
-
         ToolTipManager.sharedInstance().setInitialDelay(GUIP.getTooltipDelay());
         if (GUIP.getTooltipDismissDelay() > 0) {
             ToolTipManager.sharedInstance().setDismissDelay(GUIP.getTooltipDismissDelay());
+        } else {
+            ToolTipManager.sharedInstance().setDismissDelay(Integer.MAX_VALUE);
```

Comment:
DismissDelay has to be a positive number, there were a few places putting -1 as the "value" for set dismiss delay, but that made it instantaneous. So this check looks if the value is bigger than 0, otherwise it will set the dismiss delay to a very large number, making it "effectively" present in there while the user doesnt dismiss it by moving the mouse or something like this.